src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -114,79 +114,98 @@ def create_appliance(resource_group_name,
             raise CLIError('--plan-name, --plan-product, --plan-publisher and \
             --plan-version are all required if kind is MarketPlace')
         else:
-            appliance.plan = Plan(plan_name, plan_publisher, plan_product, plan_version)
+            application.plan = Plan(plan_name, plan_publisher, plan_product, plan_version)
 
-    applianceParameters = None
+    applicationParameters = None
 
     if parameters:
         if os.path.exists(parameters):
-            applianceParameters = get_file_json(parameters)
+            applicationParameters = get_file_json(parameters)
         else:
-            applianceParameters = shell_safe_json_parse(parameters)
+            applicationParameters = shell_safe_json_parse(parameters)
 
-    appliance.parameters = applianceParameters
+    application.parameters = applicationParameters
 
-    return racf.appliances.create_or_update(resource_group_name, appliance_name, appliance)
+    return racf.applications.create_or_update(resource_group_name, application_name, application)
 
 
-def show_appliance(resource_group_name=None, appliance_name=None):
+def show_application(resource_group_name=None, application_name=None):
     """ Gets a managed application.
     :param str resource_group_name:the resource group name
-    :param str appliance_name:the managed application name
+    :param str application_name:the managed application name
     """
     racf = _resource_managedapps_client_factory()
-    return racf.appliances.get(resource_group_name, appliance_name)
+    return racf.applications.get(resource_group_name, application_name)
 
 
-def show_appliancedefinition(resource_group_name=None, appliance_definition_name=None):
+def show_applicationdefinition(resource_group_name=None, application_definition_name=None):
     """ Gets a managed application definition.
     :param str resource_group_name:the resource group name
-    :param str appliance_definition_name:the managed application definition name
+    :param str application_definition_name:the managed application definition name
     """
     racf = _resource_managedapps_client_factory()
-    return racf.appliance_definitions.get(resource_group_name, appliance_definition_name)
+    return racf.application_definitions.get(resource_group_name, application_definition_name)
 
 
-def create_appliancedefinition(resource_group_name,
-                               appliance_definition_name, location,
-                               lock_level, package_file_uri, authorizations,
-                               description, display_name, tags=None):
+def create_applicationdefinition(resource_group_name,
+                                 application_definition_name, location,
```

Comment:
you should default the location to the resource group's location, rather make it manadatory

Reply:
Done